api4/post.go

```diff
@@ -127,6 +128,39 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(list.ToJson()))
 }
 
+func getFlaggedPostsForUser(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if c.Params.UserId != c.Session.UserId && !app.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_ROLES) {
```

Comment:
Can we use the `SessionHasPermissionToUser()` check instead? It should cover both cases